modules/miner/miner.go

```diff
@@ -194,10 +199,15 @@ func New(cs modules.ConsensusSet, tpool modules.TransactionPool, w modules.Walle
 // Close terminates all ongoing processes involving the miner, enabling garbage
 // collection.
 func (m *Miner) Close() error {
+	if err := m.tg.Stop(); err != nil {
```

Comment:
`m.tg.Stop()` depends on the cpu miner being turned off to close correctly. To ensure that, you should set `m.miningOn = false` before calling `m.tg.Stop()`. You should also write a test that starts the miner, turns on cpu mining, and then calls `Close` before turning off cpu mining. If all goes well, the cpu miner should close before `Stop()` returns. You can check that the cpu miner has closed by looking at the `m.mining` variable (not to be confused with `m.miningOn` - `m.mining` tells you whether or not the miner is *actually* mining, and `m.miningOn` tells you whether or not the miner is *supposed* to be mining.